packages/server/lib/util/resolve.js

```diff
@@ -15,13 +14,9 @@ module.exports = {
     }
 
     try {
-      const options = {
-        basedir: projectRoot,
-      }
+      debug('resolving typescript for %o', projectRoot)
 
-      debug('resolving typescript with options %o', options)
-
-      const resolved = resolve.sync('typescript', options)
+      const resolved = require.resolve('typescript', { paths: [projectRoot] })
```

Comment:
I think the problem is that `require.resolve` here is still not the patched `require.resolve` that Yarn v2 expects, since we are running Cypress inside of Cypress's own Electron-Node instance.

Maybe the solution is to fall back to trying to use `@yarnpkg/pnp` to discover the path if `require.resolve('typescript')` fails? https://yarnpkg.com/advanced/pnpapi#resolvetounqualified

Reply:
This code actually works fine for me, my coworkers, and our CI. Could the cause be that your e2e test isn't executing cypress from yarn like you're supposed to (`yarn cypress ...`)? I believe that's where it injects its custom resolver. It would also probably be pretty difficult to detect which package manager you are using if you're executing outside of your package manager's environment

Reply:
Oh, I didn't realize yarn v2 required launched stuff like that, it will probably pass once the test is updated then

Reply:
@blake-transcend I updated the spec to use `yarn node` to launch `cypress run` via the CLI, but it still seems to fail with that same error. Since I'm not super familiar with how to test this (even manually), do you want to take a try at writing a test for this?

Reply:
Sure I can try with your test